source/components/dialog/dialog.ts

```diff
@@ -52,18 +54,35 @@ export class DialogComponent extends FormComponent {
 			submitAndClose: () => this.submitAndClose(),
 			size: this.size,
 		});
+		const inactive$ = new Subject()
+		this.dialogRoot.openDialog.takeUntil(inactive$).subscribe(() => {
```

Comment:
I don't quite follow the logic behind these, if the dialogRoot. openDialog or closeDialog observables emit a value, then inactive$ emits a value and it completes the source. Could we just remove the inactive$ observable and replace  `takeUntil` with `take(1)`?

Reply:
I think the only think takeUntil was doing here instead of take(1) was just that it only takes the first of either of these streams. Anyway it's a moot point since I closed the PR and am trying a different approach

Reply:
Replaced by https://github.com/RenovoSolutions/TypeScript-Angular-Components/pull/425

Reply:
Oh IC, since inactive$ was emitting it would close both streams after one of them had emitted a value. 

As an alternative, you could have probably used the `Observable.race`  either way, the replacement is way better 👍